src/uwtools/drivers/forecast.py

```diff
@@ -162,12 +162,28 @@ def job_card(self):
         Turns the resources config object into a batch card for the configured Task.
         """
 
-    def run(self):
-        """
-        Runs the forecast executable with the namelist file and staged input files.
-
-        This will both build the executable and run it.
-        """
+    def run(
+        self,
+        exec_name: str,
+        export: Optional[str] = None,
+        mpi: Optional[str] = None,
+        node_arg: Optional[str] = None,
+        pe_member01: Optional[int] = None,
+        num_of_procs: Optional[int] = None,
+        oversubscribe: Optional[str] = None,
+        procs_per_nodes: Optional[str] = None,
+        ppn: Optional[int] = None,
+        cpu_bind: Optional[str] = None,
+        core: Optional[str] = None,
+        depth: Optional[str] = None,
+        omp_num_threads: Optional[int] = None,
+    ) -> str:
+        """
+        Builds the forecast executable with the user provided run commands.
+
+        This will build the executable.
+        """
+        return (" ".join(str(i) if i is not None else "" for i in vars().values())).strip()
```

Comment:
I wonder if `str(i).strip()` might be nice instead of the trailing `.strip()` -- if all the individual components are first stripped of leading/trailing whitespace, does that take care of the final string, too? If so, that would also clean up any unintentional leading/trailing whitespace that users might pass as part of values.

Reply:
We removed strip altogether and cleaned up the leading/trailing whitespaces.